packages/design-system/ui/src/hooks/use-prompt/use-prompt.tsx

```diff
@@ -3,11 +3,20 @@
 import * as React from "react"
 import { createRoot } from "react-dom/client"
 import { RenderPrompt, RenderPromptProps } from "./render-prompt"
+import { useRef } from "react"
 
 type UsePromptProps = Omit<RenderPromptProps, "onConfirm" | "onCancel" | "open">
 
-const usePrompt = () => {
+const usePrompt = ({ singleton = false }: { singleton?: boolean } = {}) => {
+  const flag = useRef(false)
+
   const prompt = async (props: UsePromptProps): Promise<boolean> => {
+    if (singleton && flag.current) {
```

Comment:
LGTM, but should we perhaps just change it so this is the default behaviour, without needing to do `singleton: true`? I can't really come up with any cases where you would ever want to render multiple prompts since they stack on top of each other, making everyone except the top one visible.

Reply:
Yup, makes more sense for this to be the default. Will push a change.